tests/test_logger.py

```diff
@@ -0,0 +1,86 @@
+"""
+Tests for the ralph.logger module
+"""
+
+from pathlib import Path
+
+from click.testing import CliRunner
+
+import ralph.logger
+from ralph.cli import cli
+from ralph.exceptions import ConfigurationException
+
+
+# pylint: disable=invalid-name, unused-argument
+def test_logger_exists(fs, monkeypatch):
+    """Test the logging system when a correct configuration is provided"""
+
+    mock_default_config = {
+        "version": 1,
+        "propagate": True,
+        "formatters": {
+            "ralph_format": {
+                "format": "%(asctime)-23s %(levelname)-8s %(name)-8s %(message)s"
+            },
+        },
+        "handlers": {
+            "ralph_stderr": {
+                "class": "logging.StreamHandler",
+                "level": "INFO",
+                "stream": "ext://sys.stderr",
+                "formatter": "ralph_format",
+            },
+        },
+        "loggers": {
+            "ralph": {
+                "handlers": ["ralph_stderr"],
+                "level": "INFO",
+            },
+        },
+    }
+
+    Path("dev").mkdir()
+    Path("app/.ralph").mkdir(parents=True)
+
+    monkeypatch.setattr(ralph.logger, "LOG_CONFIG", mock_default_config)
+
+    runner = CliRunner()
+    result = runner.invoke(
+        cli,
+        ["push", "-b", "fs", "test_file"],
+        input="test input",
+    )
+
+    assert result.exit_code == 0
+
+    assert "Pushing archive test_file to the configured fs backend" in result.output
```

Comment:
You have access to `stderr` in the result: https://click.palletsprojects.com/en/7.x/api/#click.testing.Result

Could be worth testing you are using the expected stream?

Reply:
Unfortunately cli runner does not capture stdout and stderr separately even though it is specified in the documentation. I don't know any workaround to this